machines/vic/install.go

```diff
@@ -0,0 +1,309 @@
+		log.Errorf("-iStore Image datastore name must be specified")
+		flag.Usage()
+	}
+
+	if data.cert != "" && data.key == "" {
+		log.Errorf("key cert should be specified at the same time")
+	}
+	if data.cert == "" && data.key != "" {
+		log.Errorf("key cert should be specified at the same time")
+	}
+
+	if data.externalNetwork == "" {
+		data.externalNetwork = "VM Network"
+	}
+
+	if data.bridgeNetwork == "" {
+		data.bridgeNetwork = data.displayName
+	}
+
+	if len(data.displayName) > MaxDisplayNameLen {
+		log.Fatalf("Display name %s exceeds the permitted 31 characters limit. Please use a shorter -name parameter", data.displayName)
+	}
+
+	data.osType = "linux"
+	data.timeout = 3 * time.Minute
+	data.logfile = "install.log"
+
+	data.conf = config.NewConfig()
+	data.conf.NumCPUs = 1
+	data.conf.MemoryMB = 2048
+	data.conf.DisplayName = data.displayName
+	data.conf.DatacenterName = strings.Split(data.computeResource, "/")[1]
+	data.conf.ClusterPath = strings.Split(data.computeResource, "/Resources")[0]
+
+	if data.conf.DatacenterName == "" || data.conf.ClusterPath == "" {
+		log.Fatalf("Wrong compute resource path %s", data.computeResource)
+	}
+
+	data.conf.ResourcePoolPath = data.computeResource
+	data.conf.ImageDatastoreName = data.imageDatastore
+	data.conf.ImageStorePath = fmt.Sprintf("/%s/datastore/%s", data.conf.DatacenterName, data.imageDatastore)
+	data.conf.ExtenalNetworkPath = fmt.Sprintf("/%s/network/%s", data.conf.DatacenterName, data.externalNetwork)
+	data.conf.ExtenalNetworkName = data.externalNetwork
+	data.conf.BridgeNetworkPath = fmt.Sprintf("/%s/network/%s", data.conf.DatacenterName, data.bridgeNetwork)
+	data.conf.BridgeNetworkName = data.bridgeNetwork
+	if data.managementNetwork != "" {
+		data.conf.ManagementNetworkPath = fmt.Sprintf("/%s/network/%s", data.conf.DatacenterName, data.managementNetwork)
+		data.conf.ManagementNetworkName = data.managementNetwork
+	}
+	//prompt for passwd if not specified
+	if data.passwd == "" {
+		log.Print("Please enter ESX or vCenter password: ")
+		b, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if err != nil {
+			log.Fatalf("Failed to read password from stdin: %s", err)
+		}
+		data.passwd = string(b)
+	}
+	data.conf.TargetPath = fmt.Sprintf("%s:%s@%s/sdk", data.user, data.passwd, data.target)
+}
+
+func validateConfiguration() error {
+	if err := loadCertificate(); err != nil {
+		return err
+	}
+
+	return data.conf.ValidateConfiguration()
+}
+
+func loadCertificate() error {
+	var keypair *Keypair
+	if data.cert != "" && data.key != "" {
+		log.Infof("Loading certificate/key pair - private key in %s", data.key)
+		keypair = NewKeyPair(false, data.key, data.cert)
+	} else if data.tlsGenerate {
+		data.key = fmt.Sprintf("./%s-key.pem", data.displayName)
+		data.cert = fmt.Sprintf("./%s-cert.pem", data.displayName)
+		log.Infof("Generating certificate/key pair - private key in %s", data.key)
+		keypair = NewKeyPair(true, data.key, data.cert)
+	}
+	if keypair == nil {
+		return nil
```

Comment:
log Message: "Configuring without TLS - to enable use -gen or -key/-cert parameters"